src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -14,6 +14,11 @@ public class StringFunctions {
 
     public static final String reverseWithStringConcat(String string) {
         String output = new String();
+
+        if(string==null || string.isEmpty()){  //added in a change such that it handles the null case and if it is empty 
```

Comment:
Line 18 added in an if statement that checks if the string parameter passed into the function is null or if it is empty. If so, the program returns an empty string.